crates/sparrow-syntax/src/syntax/fenl_type.rs

```diff
@@ -16,8 +18,8 @@ pub enum FenlType {
     // TODO: Rename concrete?
     /// A specific Arrow DataType.
     Concrete(DataType),
-    /// A generic type with the given type constraint.
-    Generic(TypeConstraint),
+    /// A generic type with the given type variable.
+    Generic(TypeVariable),
```

Comment:
For a follow-up fixing naming, I think the `Generic` should be a `TypeRef` perhaps?